src/Collection/Collection.service.ts

```diff
@@ -510,6 +510,20 @@ export class CollectionService {
   }
 
   public async isDCLPublished(contractAddress: string) {
+    // First check if collection exist in the database and has no items with a blockhain item id already
+    const dbCollection = await Collection.findOne<CollectionAttributes>({
+      contract_address: contractAddress,
+    })
+    if (!dbCollection) {
+      return false
+    }
+    const collectionItems = await Item.find<ItemAttributes>({
+      collection_id: dbCollection.id,
+    })
+    if (collectionItems.some((item) => item.blockchain_item_id !== null)) {
+      return true
+    }
```

Comment:
What if you create a new query in the `Items.model` file that goes through the items that belong to a collection to see if there's one published? This change will greatly improve the performance of the operation.

Reply:
done ✅